zookeeper-server/src/test/java/org/apache/zookeeper/server/persistence/FileTxnSnapLogMetricsTest.java

```diff
@@ -43,38 +40,20 @@
 
     private static final Logger LOG = LoggerFactory.getLogger(FileTxnSnapLogMetricsTest.class);
 
-    CountDownLatch allCreatedLatch;
-
-    private class MockWatcher implements Watcher {
-
-        @Override
-        public void process(WatchedEvent e) {
-            LOG.info("all nodes created");
-            allCreatedLatch.countDown();
-        }
-
-    }
-
     @Test
     public void testFileTxnSnapLogMetrics() throws Exception {
-        SyncRequestProcessor.setSnapCount(100);
+        // disable automatic snapshot taking to leave writes in txn log
+        SyncRequestProcessor.setSnapCount(Integer.MAX_VALUE);
 
```

Comment:
We should reset this in a finally block.
That was a pre existing problem

Reply:
Not sure, but I think almost all tests run and should run in dedicated jvm(aka. false `reuseForks` in `zookeeper-server` module). This test is not an exception. Restoring a primitive value has no means in this case.

Reply:
I'm not exactly sure how we run the tests... sometimes I even run them in IntelliJ during development. I think usually setting back all "static stuff" we changed in a test is a good practice.

Reply:
Cleanup is always a complicated problem. No ones run in identical environment. 

> setting back all "static stuff" we changed in a test is a good practice.

It is actually hard to follow. Given `SyncRequestProcessor.setSnapCount` as an example, there are 20+ usages in test, only `RecoveryTest.testRecovery` restores.

Personally, I treat all tests as oneshot and resort to junit's `@Rule`(eg. `@TempDir`) or similars to cleanup persistent resources.

Anyway, I will add a `@AfterEach` to restore it back in next fixup. 😄